tests/integration_tests/test_torchscript.py

```diff
@@ -422,10 +422,20 @@ def validate_torchscript_outputs(tmpdir, config, backend, training_data_csv_path
             assert output_name in feature_outputs
             output_values = feature_outputs[output_name]
             if isinstance(output_values, list):
-                # Strings should match exactly
-                assert np.all(
-                    output_values == output_values_expected
-                ), f"feature: {feature_name}, output: {output_name}"
+                if isinstance(output_values[0], torch.Tensor):
+                    # If list of numeric values (e.g., probabilities of SetFeature), compare as numpy arrays.
+                    for i in range(len(output_values)):
+                        assert (
+                            output_values[i].shape == output_values_expected[i].shape
+                        ), f"feature: {feature_name}, output: {output_name}"
+                        assert np.allclose(
+                            output_values[i], output_values_expected[i]
+                        ), f"feature: {feature_name}, output: {output_name}"
```

Comment:
nit: This level of nesting makes me sweat! Could we abstract this into a separate function, i.e. `assert_set_feature_probabilities` or `assert_number_set_feature`?

Reply:
Good point! Refactored.